src/main/java/jewellery/inventory/service/OrganizationService.java

```diff
@@ -63,4 +118,85 @@ private Organization getOrganization(UUID id) {
+    if (!hasManageUsersPermission(currentUser, organization)) {
+      throw new UserNotHaveUserPermissionException(currentUser.getId(), organization.getId());
+    }
+    logger.debug(
+        "User permission validation successful. User ID: {}, Organization ID: {}",
+        currentUser.getId(),
+        organization.getId());
+  }
+
+  private void validateUserInOrganization(Organization organization) {
+    User currentUser = userService.getUser(authService.getCurrentUser().getId());
+    boolean isUserInOrganization =
+        organization.getUsersInOrganization().stream()
+            .anyMatch(userInOrganization -> userInOrganization.getUser().equals(currentUser));
+
+    if (!isUserInOrganization) {
+      throw new UserIsNotPartOfOrganizationException(currentUser.getId(), organization.getId());
+    }
+    logger.debug(
+        "User permission validation successful. User ID: {}, Organization ID: {}",
+        currentUser.getId(),
+        organization.getId());
+  }
+
+  private boolean hasManageUsersPermission(User user, Organization organization) {
+    return organization.getUsersInOrganization().stream()
+        .anyMatch(
+            userInOrganization ->
+                userInOrganization.getUser().equals(user)
+                    && userInOrganization
+                        .getOrganizationPermission()
+                        .contains(OrganizationPermission.MANAGE_USERS));
+  }
+
+  private UserInOrganization createUserInOrganization(
+      UserInOrganizationRequestDto requestDto, Organization organization) {
+    UserInOrganization userInOrganization = new UserInOrganization();
+    userInOrganization.setUser(userService.getUser(requestDto.getUserId()));
+    userInOrganization.setOrganization(organization);
+    userInOrganization.setOrganizationPermission(requestDto.getOrganizationPermission());
+    userInOrganizationRepository.save(userInOrganization);
+    logger.debug(
+        "Successfully created user in organization. User ID: {}, Organization ID: {}, Organization Permission: {}",
+        userInOrganization.getUser().getId(),
+        userInOrganization.getOrganization().getId(),
+        userInOrganization.getOrganizationPermission());
+    return userInOrganization;
+  }
+
+  private void addUserToOrganization(
+      UserInOrganization userInOrganization, Organization organization) {
+    organization.getUsersInOrganization().add(userInOrganization);
+    logger.info(
+        "Successfully added user in the organization. Organization ID: {}, User ID: {}",
+        organization.getId(),
+        userInOrganization.getUser().getId());
+    organizationRepository.save(organization);
+  }
+
+  private void changeUserPermissionInOrganization(
+      Organization organization,
+      User userForUpdate,
+      List<OrganizationPermission> organizationPermissionList) {
+    organization
+        .getUsersInOrganization()
+        .forEach(
```

Comment:
Let's not use forEach here. Use somthing like findFirst for the ID and change that specific userInOrganization